src/imgui/ImGuiMachine.cc

```diff
@@ -46,8 +46,69 @@ ImGuiMachine::ImGuiMachine(ImGuiManager& manager_)
 	, setupFileList("setup", Reactor::SETUP_EXTENSION, Reactor::SETUP_DIR)
 	, confirmDialog("Confirm##setup")
 {
-	setupFileList.selectAction = [&](const FileListWidget::Entry& entry) {
-		manager.executeDelayed(makeTclList("setup", entry.displayName));
+
+	setupFileList.drawAction = [&] {
+		im::Table("table", 2, ImGuiTableFlags_BordersInnerV, [&]{
+			if (ImGui::TableNextColumn()) {
+				setupFileList.drawTable();
+			}
+			if (ImGui::TableNextColumn()) {
+				if (previewSetup.motherBoard) {
+					showSetupOverview(*previewSetup.motherBoard);
+					ImGui::Separator();
+					if (ImGui::Button("Run!")) {
+						manager.executeDelayed([&previewSetup = previewSetup, &manager = manager] {
+							manager.getReactor().switchMachineFromSetup(previewSetup.fullName);
+						});
+					}
+				} else {
+					if (previewSetup.lastExceptionMessage.empty()) {
+						ImGui::TextUnformatted("Nothing to preview...");
+					} else {
+						im::StyleColor(ImGuiCol_Text, getColor(imColor::ERROR), [&]{
+							ImGui::TextUnformatted(tmpStrCat("Setup ", previewSetup.displayName, " cannot be loaded:"));
```

Comment:
Detail: combination of `TextUnformatted` and `(tmp)StrCat()` can be replaced with `ImGui::StrCat()`.
You can consider including the next line into this one

Reply:
OK done